lib/testers/rule-tester.js

```diff
@@ -396,10 +396,14 @@ class RuleTester {
          * @private
          */
         function assertASTDidntChange(beforeAST, afterAST) {
-            if (!lodash.isEqual(beforeAST, afterAST)) {
 
-                // Not using directly to avoid performance problem in node 6.1.0. See #6111
+            // Feature detect the Node.js implementation.
+            if (util.isDeepStrictEqual) {
                 assert.deepStrictEqual(beforeAST, afterAST, "Rule should not modify AST.");
```

Comment:
I'm confused about this:

* It looks like we're using feature detection by checking for `util.isDeepStrictEqual`, but then we're never actually calling `util.isDeepStrictEqual`. Is that intentional?
* Why is feature detection necessary here? Is there an advantage to using Node's implementation when available, rather than always using `lodash.isEqual`? I'm not sure the behaviors of `lodash.isEqual` and `util.isDeepStrictEqual` are identical, so it seems like using feature detection could lead to cases where the behavior is slightly different depending on the Node version, which seems undesirable.

Reply:
Well, in this case `util.isDeepStrictEqual` could also directly be used. About different comparisons in different versions: that is also the case when directly using the Node.js assert version as the implementation improved over time. I personally do not have a strong opinion about it. lodash was introduced because of a bad algorithm that got fixed a while ago in `deepStrictEqual`.